### PR TITLE
Switching autoscaler-webhook to utilize distroless as base Image

### LINKED ISSUE #909: Should we rebase images from alpine to gcr.io/distroless/static ?
Kubernetes has moved to using gcr.io/distroless/static for most base images (see https://github.com/kubernetes/enhancements/blob/master/keps/sig-release/20190316-rebase-images-to-distroless.md#implementation-history). We should investigate whether it makes sense to move from alpine to gcr.io/distroless/static for Agones.

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug

/kind cleanup

> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #909

**Special notes for your reviewer**:

### CODE REVIEW

File: examples/autoscaler-webhook/Dockerfile [hunk 1]
```diff
@@ -22,7 +22,7 @@ RUN go mod init && go mod tidy
 RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o server .
 
 # Create the final image that will run the webhook server for FleetAutoscaler webhook policy
-FROM alpine:3.14
+FROM gcr.io/distroless/static-debian11:nonroot
 RUN adduser -D -u 1000 server
```
Comment (markmandel) [inline examples/autoscaler-webhook/Dockerfile:26]:
```suggestion
WORKDIR /
```

No need for this anymore, and we move everything into root.

File: examples/autoscaler-webhook/Dockerfile [hunk 2]
```diff
@@ -22,7 +22,7 @@ RUN go mod init && go mod tidy
...
 
 COPY --from=builder /go/src/autoscaler-webhook \
```
Comment (markmandel) [inline examples/autoscaler-webhook/Dockerfile:28]: change this to copy to `/server`

File: examples/autoscaler-webhook/Dockerfile [hunk 3]
```diff
@@ -31,5 +31,5 @@ COPY --from=builder /go/src/autoscaler-webhook \
 RUN chown -R server /home/server && \
```
Comment (markmandel) [inline examples/autoscaler-webhook/Dockerfile:31]: This can all go away.

File: examples/autoscaler-webhook/Dockerfile [hunk 4]
```diff
@@ -31,5 +31,5 @@ COPY --from=builder /go/src/autoscaler-webhook \
...
     chmod o+x /home/server/server
 
-USER 1000
-ENTRYPOINT /home/server/server
+USER nonroot:nonroot
+ENTRYPOINT ["/home/server/server"]
```
Comment (markmandel) [inline examples/autoscaler-webhook/Dockerfile:35]:
```suggestion
ENTRYPOINT ["/server"]
```

**Review by markmandel — COMMENTED** [review]
Good start!

Can you also have a look through our docs to see what references the `autoscaler-webhook` container image, and update that as well?

We'll also need to manually test to make sure it still works please.

**Comment by Kalaiselvi84** [comment]
> Can you also have a look through our docs to see what references the `autoscaler-webhook` container image, and update that as well?
Do we have a need to update all files currently using "autoscaler-webhook 0.6" to instead use "autoscaler-webhook 0.7"?

**Comment by markmandel** [comment]
> Do we have a need to update all files , excluding release blogs, that currently using "autoscaler-webhook 0.6" to instead use "autoscaler-webhook 0.7"?

Yes please!

**Comment by Kalaiselvi84** [comment]
> We'll also need to manually test to make sure it still works please.

could you please provide more context?

**Comment by markmandel** [comment]
> could you please provide more context?

Ah yes - excellent question!

We use this image in this quickstart: https://agones.dev/site/docs/getting-started/create-webhook-fleetautoscaler/

So if you can go through this quickstart, but utilise your new image rather than the old one, that would be perfect.

**Review by markmandel — APPROVED** [review]
Chatted with @Kalaiselvi84 offline. Tested, and looks good 👍🏻

**Comment by markmandel** [comment]
```
--- FAIL: TestAllocatorAllocateOnGameServerUpdateError (19.31s)
    allocator_test.go:499: 
        	Error Trace:	/go/src/agones.dev/agones/pkg/gameserverallocations/allocator_test.go:499
        	Error:      	An error is expected but got nil.
        	Test:       	TestAllocatorAllocateOnGameServerUpdateError
```

flake. I thought I fixed this.

**Comment by markmandel** [comment]
@Kalaiselvi84 is the image pushed yet? If not, that would explain why the e2e tests are failing 😄

**Comment by Kalaiselvi84** [comment]
> @Kalaiselvi84 is the image pushed yet? If not, that would explain why the e2e tests are failing 😄

I used `go mod init` to create the go.mod and go.sum files, but unfortunately, the issue persists when attempting to run the `make push` command.

**Pull request merged**